src/coins.js

```diff
@@ -18,6 +19,10 @@ coins.isBitcoinCash = function (network) {
   return typeforce.value(coins.BCH)(network.coin)
 }
 
+coins.isBitcoinSV = function (network) {
```

Comment:
I don't have much context ... but this pattern seems like it could be improved (and won't scale gracefully if we add X more UTXO coins).

It looks like the only place this is being used is `src/transaction_builder.js::TransactionBuilder.prototype.sign()` to decide which hash signature function the coin should be mapped to based on the coin in question.

Probably out of the scope of this change -- but this architecture is doesn't feel like it's going to scale particularly well the more UTXO coins we add -- each with their own hash signature logic. Do we have the bandwidth to bite the bullet now and refactor this to either:

* Move the config of coin specific logic to this `coins.js` file along the lines of:
```javascript
const COIN_CONFIG = {
  BCH: {
    "symbol": "bch",
    "name": "Bitcoin Cash",
    "hash_sig_fcn": hashForCashSignature,
     etc...,
  },
  ZEC: {
    "symbol": "zec",
    "name": "Zcash",
    "hash_sig_fcn": hashForZcashSignature,
     etc...,
  },
  ....
}
```
* Or, separate logic in `transaction_builder.js` to more of a class based approach (written in python):
```python
class BaseTransactionBuilder(object):
    def sign(self, vin, keyPair, redeemScript, hashType, witnessValue, witnessScript):
        """Core transaction signing logic."""
        # Execute core signing logic that each coin requires here

        # Add hook to allow coin-specific subclasses to implement their own 
        # "get signature hash" logic.
        signatureHash = self.getSignatureHash(vin, input.signScript, witnessValue, hashType)

        # Continue with signing logic

    def getSignatureHash(self, vin, signScript, witnessValue, hashType):
        """Base get signature hash."""
        # core logic (used by BTC and anything else that isn't custom)


class BSVTransactionBuilder(BaseTransactionBuilder):    
    def getSignatureHash(self, vin, signScript, witnessValue, hashType):
        """Custom get signature hash fcn for BSV and BCH."""
        # custom BCH + BSV sig hash logic.

######## USAGE #########
COIN_MAPPING = {
  BTC: {
     "transactionBuilderClass": BaseTransactionBuilder,
  },
  BCH: {
      "transactionBuilderClass": BSVTransactionBuilder
  }
}

# Use a coin to get all necessary mapping data from a single config ^^
coinMapping = COIN_MAPPING[BCH]
transactionBuilder = new coinMapping["transactionBuilderClass"](args)
transactionBuilder.sign()  # based on coin type, the appropriate tx builder class is called and used to implement signing.
```


Continuing to add to a potentially confusing conditional doesn't feel like a pragmatic approach.

Reply:
This is part of the technical debt for this lib. Ideally, we would implement inheritance with a UTXO base class, but that is a lot of work and out of scope for this diff. Also, this refactor would include breaking changes not just to our platform, but to anyone out there using this lib.

I started a design doc some time ago but haven't finished it yet.